client-js/app/elements/labrad-plot1d.ts

```diff
@@ -0,0 +1,255 @@
+import 'd3';
+
+var COLOR_LIST = ['red', 'blue', 'yellow', 'green', 'magenta'];
+
+export interface Plot {
+  addData(data: Array<Array<number>>): void
+}
+
+@component('labrad-plot1d')
+export class Plot1D extends polymer.Base implements Plot {
+
+  @property({type: String, value: ''})
+  xLabel: string;
+
+  @property({type: String, value: ''})
+  yLabel: string;
+
+  private data: Array<Array<number>> = []
+  private lastData: Array<number> = null;
+
+  private numTraces: number = 0;
+  private svg: any;
+  private chartBody: any;
+  private clip: any;
+  private xAxis: any;
+  private yAxis: any;
+  private xScale: any;
+  private yScale: any;
+  private line: any;
+  private zoom: any;
+  private limits = {xMin: 0, xMax: 1, yMin: 0, yMax: 1};
+  private dataLimits = {xMin: NaN, xMax: NaN, yMin: NaN, yMax: NaN};
+  private margin: {top: number; right: number; bottom: number; left: number};
+
+  private plot: any;
+
+  attached() {
+    this.redraw();
+    window.addEventListener('resize', (event) => this.redraw());
+  }
+
+  @observe('xLabel')
+  xLabelChanged(newLabel: string, oldLabel: string) {
+    if (this.svg) {
+      this.svg.select('#x-label').text(newLabel);
+    }
+  }
+
+  @observe('yLabel')
+  yLabelChanged(newLabel: string, oldLabel: string) {
+    if (this.svg) {
+      this.svg.select('#y-label').text(newLabel);
+    }
+  }
+
+  private redraw() {
+    var rect = this.getBoundingClientRect();
+    while (this.firstChild) {
+      this.removeChild(this.firstChild);
+    }
+    this.createPlot(Math.max(rect.width, 320), Math.max(rect.height, 200));
+    this.plotData(this.data);
+  }
+
+  private createPlot(totWidth: number = 1000, totHeight: number = 500): void {
+    var p = this;
+    p.limits = {xMin: 0, xMax: 100, yMin: 0, yMax: 100};
+    p.margin = {top: 50, right: 50, bottom: 50, left: 40};
+
+    var width = totWidth - p.margin.left - p.margin.right;
+    var height = totHeight - p.margin.top - p.margin.bottom;
+
+    p.xScale = d3.scale.linear()
+            .domain([p.limits.xMin, p.limits.xMax])
+            .range([0, width]);
+
+    p.yScale = d3.scale.linear()
+            .domain([p.limits.yMin, p.limits.yMax])
+            .range([height, 0]);
+
+    p.xAxis = d3.svg.axis()
+            .scale(p.xScale)
+            .orient('bottom')
+            .tickSize(-height);
+
+    p.yAxis = d3.svg.axis()
+            .scale(p.yScale)
+            .orient('left')
+            .ticks(5)
+            .tickSize(-width);
+
+    p.line = d3.svg.line()
+            .x((d) => p.xScale(d[0]))
+            .y((d) => p.yScale(d[1]));
+
+    p.zoom = d3.behavior.zoom()
+            .x(p.xScale)
+            .y(p.yScale)
+            .on('zoom', p.handleZoom.bind(p));
+
+    // plot area
```

Comment:
`/** */` is for jsdoc comments on visible parts of the api (e.g. classes, methods, or public attributes). Inline comments should use `//`.